RGB-BRIGHTNESS/APP/app.c

```diff
@@ -75,56 +91,55 @@ en_app_error_t app_init(void)
     en_app_error_t en_app_error_retval = APP_OK;
     en_btn_status_code_t_ en_btn_status_code = BTN_STATUS_OK;
     en_led_error_t_ en_led_error = LED_OK;
-    en_systick_error_t en_systick_error = ST_OK;
-
-    /* init RGB LED */
-
-    // init RED LED
-    en_led_error = led_init(RED_LED_PORT, RED_LED_PIN);
-    if(LED_OK != en_led_error) en_app_error_retval = APP_FAIL;
 
-    // init blue led
-    en_led_error = led_init(GREEN_LED_PORT, GREEN_LED_PIN);
-    if(LED_OK != en_led_error) en_app_error_retval = APP_FAIL;
-
-    // init green LED
-    en_led_error = led_init(BLUE_LED_PORT, BLUE_LED_PIN);
-    if(LED_OK != en_led_error) en_app_error_retval = APP_FAIL;
+    /* init RGB LEDS */
+    for (int i = ZERO; i < RGB_LEDS_COUNT; ++i) {
+        en_led_error = led_init(gl_st_app_leds[i].en_led_port, gl_st_app_leds[i].en_led_pin);
+        if(LED_OK != en_led_error) en_app_error_retval = APP_FAIL;
+    }
 
     // init button
     en_btn_status_code = btn_init(&gl_st_user_btn_cfg);
     if(BTN_STATUS_OK != en_btn_status_code) en_app_error_retval = APP_FAIL;
 
-    // Init Systick
-    en_systick_error = systick_init(&gl_st_systick_cfg_0);
-    if(ST_OK != en_systick_error) en_app_error_retval = APP_FAIL;
+    /* Init PWM module */
+    en_pwm_error_t en_pwm_status = pwm_init();
+    if(PWM_OK != en_pwm_status) en_app_error_retval = APP_FAIL;
 
-    en_systick_error = systick_set_callback(&app_systick_cb);
-    if(ST_OK != en_systick_error) en_app_error_retval = APP_FAIL;
 
     return en_app_error_retval;
 }
 
 void app_start(void)
 {
-    while(1)
+    while(TRUE)
     {
         en_btn_state_t_ en_btn_state = BTN_STATE_NOT_PRESSED;
         btn_read(&gl_st_user_btn_cfg, &en_btn_state);
 
         if(BTN_STATE_PRESSED == en_btn_state)
         {
-            if(ALL_LEDS == gl_en_app_sub_state)
-            {
-                gl_en_app_sub_state = ALL_OFF;
-            }
-            else
-            {
-                gl_en_app_sub_state += 1;
-            }
-
             // set app state -> switching
             gl_en_app_state = SWITCHING;
+
+            switch (gl_en_app_sub_state) {
```

Comment:
Switch case should have only one break

Reply:
There was no proof of such rule in MISRA, and to be frank it isn't meaningful. that's the whole point of a switch() statement to have multiple options instead of doing too many if/else nesting.

Reply:
I don't know the exact version which mentions this rule so you have two solutions to avoid doing switch and to be compliant with this MISRA rule:

1.The easy way to replace the switch case with if condition.
2.The hard way is to use Fall-through switch cases.

Reply:
> I don't know the exact version which mentions this rule so you have two solutions to avoid doing switch and to be compliant with this MISRA rule:
> 
> 1.The easy way to replace the switch case with if condition. 2.The hard way is to use Fall-through switch cases.

Thanks again for the detailed input, sorry for the late reply,

We modified the code to comply with the rule you've mentioned, we opted out to option 1 (used if/else) instead
We also optimized the PWM driver which helped in optimizing the application flow, as we now removed the `led_interface.h` include from the APP file as the PWM driver is now fully controlling and managing the PWM signals.

Commit Ref: 5fb6fd8d07dada624f1ba7e4acb3bdfd9b4d11b0

Reply:
Great work...I re-reviewed your code again and would like to thank you for your effort.